src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -148,6 +157,100 @@
     (catch Exception e
       (log/error e "Error during garbage collection"))))
 
+;; Without atomic multi-metric updates, there will be
+;; cleaning metric races, and this doesn't work at all with multiple
+;; PDB instances in the same VM.
```

Comment:
Perhaps, mostly just put it there for pr discussion; I suspect these may be existing issues.